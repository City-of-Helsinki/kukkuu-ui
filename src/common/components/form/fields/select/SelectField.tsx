import React, { Component } from 'react';

type SelectOption = {
  label: string;
  value: string | string[] | number;
};
type SelectOptions = SelectOption[];

type Props = {
  options: SelectOptions;
  name: string;
  id: string;
};

class SelectField extends Component<Props> {
  render() {
    const { options, name, id } = this.props;

    return (
      <select name={name} id={id}>
        {options.map(selectOption => (
          <option value={selectOption.value} key={selectOption.label}>
            {selectOption.label}
          </option>
        ))}
      </select>
    );
  }
}

export default SelectField;
