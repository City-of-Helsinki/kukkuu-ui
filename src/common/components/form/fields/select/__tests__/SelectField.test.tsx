import React from 'react';
import { shallow } from 'enzyme';
import { Field } from 'formik';

import SelectField from '../SelectField';
import TestForm from '../../../../../test/TestForm';

it('renders snapshot correctly', () => {
  const props = {
    id: 'fooSelect',
    label: 'select label',
    onChange: () => jest.fn(),
    options: [
      {
        label: 'foo_label',

        value: 'foo',
      },
      {
        label: 'bar_label',
        value: 'bar',
      },
    ],
  };
  const input = shallow(
    <TestForm>{() => <Field component={SelectField} {...props} />}</TestForm>
  );
  expect(input.html()).toMatchSnapshot();
});
