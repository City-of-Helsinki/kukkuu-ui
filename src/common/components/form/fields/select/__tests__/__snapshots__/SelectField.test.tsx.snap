// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `"<div class=\\"selectWrapper\\"><label for=\\"fooSelect\\">select label</label><select id=\\"fooSelect\\" field=\\"[object Object]\\"><option selected=\\"\\" value=\\"foo\\">foo_label</option><option value=\\"bar\\">bar_label</option></select></div>"`;
